private/protocol/rest/build.go

```diff
@@ -92,7 +92,11 @@ func buildLocationElements(r *request.Request, v reflect.Value) {
 	}
 
 	r.HTTPRequest.URL.RawQuery = query.Encode()
-	updatePath(r.HTTPRequest.URL, r.HTTPRequest.URL.Path)
+	updatePath(
+		r.HTTPRequest.URL,
+		r.HTTPRequest.URL.Path,
+		r.Config.DisableRestProtocolURICleaning != nil && *r.Config.DisableRestProtocolURICleaning,
```

Comment:
can use `aws.BoolValue(r.Config.DisableRestProtocolURICleaning)` instead of needing to use `&&` here